packages/core/core-flows/src/payment-collection/steps/cancel-payment.ts

```diff
@@ -0,0 +1,56 @@
+import { IPaymentModuleService, Logger } from "@medusajs/framework/types"
+import {
+  ContainerRegistrationKeys,
+  Modules,
+  promiseAll,
+} from "@medusajs/framework/utils"
+import { createStep, StepResponse } from "@medusajs/framework/workflows-sdk"
+
+/**
+ * The data to cancel payments.
+ */
+export interface CancelPaymentStepInput {
+  /**
+   * The IDs of the payments to cancel.
+   */
+  ids: string[]
+}
+
+export const cancelPaymentStepId = "cancel-payment"
+/**
+ * This step cancels one or more authorized payments.
+ */
+export const cancelPaymentStep = createStep(
```

Comment:
thought: We typically don't send multiple third-party requests in steps, as that could lead to issues with data inconsistencies between Medusa and the third-party system.

For example, if the first payment cancellation succeeds but the second one fails, the workflow would trigger a rollback aborting the entire cancel operation. However, the first payment is still left canceled in the third-party provider.

Should we rethink this flow?

cc @carlos-r-l-rodrigues

Reply:
Seems that we been doing this already for [cancel order](https://github.com/medusajs/medusa/blob/develop/packages/core/core-flows/src/payment/steps/cancel-payment.ts#L23) so we should revisit that as well. (Also just realised this is almost identical step so will remove this one).